docs/examples/torch_distributed.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import time
+
+import torch
+
+import submitit
+
+NUM_NODES = 2
+NUM_TASKS_PER_NODE = 8
+
+
+NUM_CPUS_PER_TASK = 1
+PARTITION = "devlab"
+LOGS_DIR = "logs"
+
+
+def print_env():
+    for key in sorted(os.environ.keys()):
+        if not (
+            key.startswith(("SLURM_", "SUBMITIT_"))
+            or key in ("MASTER_ADDR", "MASTER_PORT", "RANK", "WORLD_SIZE", "LOCAL_RANK", "LOCAL_WORLD_SIZE")
+        ):
+            continue
+        value = os.environ[key]
+        print(f"{key}={value}")
+
+
+class Task:
+    def __call__(self):
+        # print_env()
+        print("exporting PyTorch distributed environment variables")
+        dist_env = submitit.helpers.TorchDistributedEnvironment().export()
+        print(f"master: {dist_env.master_addr}:{dist_env.master_port}")
+        print(f"rank: {dist_env.rank}")
+        print(f"world size: {dist_env.world_size}")
+        print(f"local rank: {dist_env.local_rank}")
+        print(f"local world size: {dist_env.local_world_size}")
+        # print_env()
+
+        # Using env:// initialization method
+        backend = torch.distributed.Backend.NCCL
+        torch.distributed.init_process_group(backend=backend)
```

Comment:
would it be cleaner to move the call of init_process_group inside the TorchDistributedEnvironment ? That way it could pass world_size and rank directly to `init_process_group` without using environment variable (that could conflict with other stuff)

Reply:
This was discussed and also suggested by @jrapin but:
- It introduces a conceptual (even if lazily imported) dependency from submitit to PyTorch instead of separating concerns between abstraction for job submission and distributed framework,
- Hiding PyTorch distributed initialization would certainly remove one line of code, but could actually increase the complexity of the codebases trying to keep a single code path working in different contexts (e.g. local, submitit job submission or slurm scripts, etc...).

So we ended up agreeing to not embed this call. If there is a strong demand, an option could then be easily introduced to also setup PyTorch distributed with some forwarded options.